qiskit_experiments/framework/backend_data.py

```diff
@@ -0,0 +1,145 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Backend data access helper class
+
+Since `BackendV1` and `BackendV2` do not share the same interface, this
+class unifies data access for various data fields.
+"""
+from qiskit.providers import BackendV1, BackendV2
+
+
+class BackendData:
```

Comment:
It is really nice to have this compatibility layer for transitioning to BackendV2. I wonder what is the life cycle of this class? Does it provide enough utility that we keep it after BackendV1 is deprecated? If not, should we try to keep the methods here as close to BackendV2 as possible so that in the future we could swap BackendData usage out for direct access of the BackendV2 instance?

Reply:
My hope is that after deprecating `BackendV1` we can stop using this class, but that's actually the reason I preferred to use static functions called with explicitly using the `BackendData` name, so it'll be easier to refactor this out. However, I can also try the `_backend_data` approach; if all goes well, the references to this field could be changed with references to the backend itself.